workspaces/adr/plugins/adr-backend/src/service/router.ts

```diff
@@ -14,18 +14,30 @@
  * limitations under the License.
  */
 
-import { CacheClient, UrlReader } from '@backstage/backend-common';
+import {
+  CacheClient,
+  createLegacyAuthAdapters,
+  PluginEndpointDiscovery,
+  UrlReader,
+} from '@backstage/backend-common';
 import { NotModifiedError, stringifyError } from '@backstage/errors';
 import express from 'express';
 import Router from 'express-promise-router';
 import { madrParser } from '../search/madrParser';
-import { LoggerService } from '@backstage/backend-plugin-api';
+import {
+  AuthService,
+  HttpAuthService,
+  LoggerService,
+} from '@backstage/backend-plugin-api';
 
 /** @public */
 export type AdrRouterOptions = {
   reader: UrlReader;
   cacheClient: CacheClient;
   logger: LoggerService;
+  discovery: PluginEndpointDiscovery;
+  httpAuth?: HttpAuthService;
+  auth?: AuthService;
```

Comment:
Following up on my other comment - these params wouldn't be necessary since you shouldn't have to interact with auth for this feature